### PR TITLE
Case 21730: Remove composite framebuffer from display plugins

### PR DESCRIPTION
**Pull request by jherico**
[21730](https://highfidelity.manuscript.com/f/cases/21730/Remove-extra-compositing-buffer-from-display-plugins)

### CODE REVIEW

File: libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp [hunk 1]
```diff
@@ -788,7 +801,12 @@ bool OpenGLDisplayPlugin::setDisplayTexture(const QString& name) {
 }
 
 QImage OpenGLDisplayPlugin::getScreenshot(float aspectRatio) const {
-    auto size = _compositeFramebuffer->getSize();
+    if (!_currentFrame || !_currentFrame->framebuffer) {
+        return QImage();
```
Comment (SamGondelman) [inline libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp]: On quest, when you're always presenting (but passing in `nullptr` to internalPresent, will this fail sometimes?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18050/HighFidelity-Beta-PR15157-4da895c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18050/HighFidelity-Beta-Interface-PR15157-4da895c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18050/HighFidelity-Beta-PR15157-4da895c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18050/HighFidelity-Beta-Interface-PR15157-4da895c.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18058/HighFidelity-Beta-PR15157-09dc885.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18058/HighFidelity-Quest-Beta-PR15157-09dc885.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18058/HighFidelity-Beta-PR15157-09dc885.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18058/HighFidelity-Beta-Interface-PR15157-09dc885.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18058/HighFidelity-Beta-PR15157-09dc885.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18058/HighFidelity-Beta-Interface-PR15157-09dc885.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18062/HighFidelity-Beta-PR15157-4f17e97.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18062/HighFidelity-Quest-Beta-PR15157-4f17e97.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18062/HighFidelity-Beta-PR15157-4f17e97.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18062/HighFidelity-Beta-Interface-PR15157-4f17e97.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18062/HighFidelity-Beta-PR15157-4f17e97.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18062/HighFidelity-Beta-Interface-PR15157-4f17e97.exe

File: libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp (hunk 1 above)
Comment (jherico) [reply libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp]: Technically, yes.... but we only pass nullptr until the first frame is ready.  So you'd have to somehow initiate a screenshot while the application is starting up, i.e. with some kind of command line argument or something.

Comment (SamGondelman) [reply libraries/display-plugins/src/display-plugins/OpenGLDisplayPlugin.cpp]: 👍

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18085/HighFidelity-Beta-PR15157-cb31140.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18085/HighFidelity-Quest-Beta-PR15157-cb31140.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18085/HighFidelity-Beta-PR15157-cb31140.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18085/HighFidelity-Beta-Interface-PR15157-cb31140.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18085/HighFidelity-Beta-PR15157-cb31140.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18085/HighFidelity-Beta-Interface-PR15157-cb31140.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18090/HighFidelity-Beta-PR15157-56d3fc4.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18090/HighFidelity-Beta-Interface-PR15157-56d3fc4.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18090/HighFidelity-Beta-PR15157-56d3fc4.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18090/HighFidelity-Quest-Beta-PR15157-56d3fc4.apk)

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18097/HighFidelity-Beta-PR15157-3a0a9c8.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18097/HighFidelity-Quest-Beta-PR15157-3a0a9c8.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18097/HighFidelity-Beta-PR15157-3a0a9c8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18097/HighFidelity-Beta-Interface-PR15157-3a0a9c8.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18097/HighFidelity-Beta-PR15157-3a0a9c8.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18097/HighFidelity-Beta-Interface-PR15157-3a0a9c8.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18152/HighFidelity-Beta-PR15157-19f856b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18152/HighFidelity-Beta-Interface-PR15157-19f856b.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18152/HighFidelity-Beta-PR15157-19f856b.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18152/HighFidelity-Quest-Beta-PR15157-19f856b.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18152/HighFidelity-Beta-PR15157-19f856b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18152/HighFidelity-Beta-Interface-PR15157-19f856b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18155/HighFidelity-Beta-PR15157-ec1268d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18155/HighFidelity-Beta-Interface-PR15157-ec1268d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18155/HighFidelity-Beta-PR15157-ec1268d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18155/HighFidelity-Beta-Interface-PR15157-ec1268d.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18159/HighFidelity-Beta-PR15157-641f9c4.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18159/HighFidelity-Beta-Interface-PR15157-641f9c4.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18159/HighFidelity-Beta-PR15157-641f9c4.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18159/HighFidelity-Quest-Beta-PR15157-641f9c4.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18162/HighFidelity-Beta-PR15157-fd65f51.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18162/HighFidelity-Beta-Interface-PR15157-fd65f51.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18159/HighFidelity-Beta-PR15157-641f9c4.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18159/HighFidelity-Beta-Interface-PR15157-641f9c4.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18162/HighFidelity-Beta-PR15157-fd65f51.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18162/HighFidelity-Quest-Beta-PR15157-fd65f51.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18162/HighFidelity-Beta-PR15157-fd65f51.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18162/HighFidelity-Beta-Interface-PR15157-fd65f51.exe

**Pull request merged**